### PR TITLE
Add additional distros/versions to btfhub archive build

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Adds the following distros to our BTF packaging, since they are now present in btfhub-archive:
- Ubuntu 16.04
- Oracle 8
- RHEL 7 and 8
- Fedora 24-28

### Motivation

Improved BTF coverage

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .gitlab/deps_build.yml [hunk 1]
```diff
@@ -144,36 +144,40 @@ build_processed_btfhub_archive:
...
-      for dir in ${dirs[@]}; do                                         \
-        platform=${dir%%/*}                                           &&\
-        mkdir -p "btfs-amd64/${platform}" "btfs-arm64/${platform}"    &&\
-        eval "mv ${dir}/x86_64/*.btf.tar.xz btfs-amd64/${platform}/"  &&\
-        eval "mv ${dir}/arm64/*.btf.tar.xz btfs-arm64/${platform}/";    \
+      for dir in ${dirs[@]}; do
+        platform=${dir%%/*}
+        mkdir -p "btfs-amd64/${platform}" "btfs-arm64/${platform}"
+        eval "mv ${dir}/x86_64/*.btf.tar.xz btfs-amd64/${platform}/"
+        eval "mv ${dir}/arm64/*.btf.tar.xz btfs-arm64/${platform}/"
```
Comment (L3n41c) [inline .gitlab/deps_build.yml:153]: Just out of curiosity, why do we need `eval` here ?
Can’t `mv` be invoked directly ?

```suggestion
        mv "${dir}"/x86_64/*.btf.tar.xz btfs-amd64/"${platform}"/
        mv "${dir}"/arm64/*.btf.tar.xz btfs-arm64/"${platform}"/
```

File: .gitlab/deps_build.yml [hunk 2]
```diff
@@ -144,36 +144,40 @@ build_processed_btfhub_archive:
...
+      done
+    # Handle the these directories separately because they don't have an arm64 version
+    - x64dirs=("amzn/2018" "fedora/24" "fedora/25" "fedora/26" "fedora/27")
+    - |
+      for xdir in ${x64dirs[@]}; do
+        platform=${xdir%%/*}
+        eval "mv ${xdir}/x86_64/*.btf.tar.xz btfs-amd64/${platform}"
```
Comment (L3n41c) [inline .gitlab/deps_build.yml:160]:
```suggestion
        mv "${xdir}"/x86_64/*.btf.tar.xz btfs-amd64/"${platform}"
```

File: .gitlab/deps_build.yml [hunk 3]
```diff
@@ -144,36 +144,40 @@ build_processed_btfhub_archive:
...
-    - mv ubuntu/18.04/x86_64/*.btf.tar.xz btfs-amd64/ubuntu/18.04/
-    - mv ubuntu/20.04/x86_64/*.btf.tar.xz btfs-amd64/ubuntu/20.04/
-    - mv ubuntu/18.04/arm64/*.btf.tar.xz btfs-arm64/ubuntu/18.04/
-    - mv ubuntu/20.04/arm64/*.btf.tar.xz btfs-arm64/ubuntu/20.04/
+    - ubuntu_dirs=("ubuntu/16.04" "ubuntu/18.04" "ubuntu/20.04")
+    - |
+      for udir in ${ubuntu_dirs[@]}; do
+        mkdir -p "btfs-amd64/${udir}" "btfs-arm64/${udir}"
+        eval "mv ${udir}/x86_64/*.btf.tar.xz btfs-amd64/${udir}/"
+        eval "mv ${udir}/arm64/*.btf.tar.xz btfs-arm64/${udir}/"
```
Comment (L3n41c) [inline .gitlab/deps_build.yml:168]:
```suggestion
        mv "${udir}"/x86_64/*.btf.tar.xz btfs-amd64/"${udir}"/
        mv "${udir}"/arm64/*.btf.tar.xz btfs-arm64/"${udir}"/
```

**Review by L3n41c — APPROVED** [review]
LGTM for the file owned by @DataDog/container-integrations.
I only have one small question for the `deps_build.yml` file: why do we need `eval` ?

**Review by usamasaqib — APPROVED** [review]

**Review by iglendd — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

File: .gitlab/deps_build.yml (hunk 1 above)
Comment (brycekahle) [reply .gitlab/deps_build.yml:153]: It fails if you remove the `eval` with a `stat` error. I think it has something to do with the wildcard. I tried to debug it, but was unsuccessful.

**Pull request merged**